tracer.go

```diff
@@ -92,13 +91,21 @@ func NewTracer(
 	}
 
 	// register default injectors/extractors
-	textPropagator := newTextMapPropagator(headersConfig, t.metrics)
-	t.injectors[opentracing.TextMap] = textPropagator
-	t.extractors[opentracing.TextMap] = textPropagator
+	textPropagator := newTextMapPropagator(getDefaultHeadersConfig(), t.metrics)
+	if _, ok := t.injectors[opentracing.TextMap]; !ok { // options has created an injector with custom header values and added it to the injector map
```

Comment:
check if injectors/extractors already set by tracer option and if now, add them.